server/src/main/java/org/opensearch/search/backpressure/trackers/HeapUsageTracker.java

```diff
@@ -117,9 +75,11 @@ public Optional<TaskCancellation.Reason> checkAndMaybeGetCancellationReason(Task
 
         double currentUsage = task.getTotalResourceStats().getMemoryInBytes();
         double averageUsage = movingAverage.getAverage();
-        double allowedUsage = averageUsage * getHeapVarianceThreshold();
+        double variance = heapVarianceSupplier.getAsDouble();
+        double allowedUsage = averageUsage * variance;
+        double threshold = heapBytesThresholdSupplier.getAsLong();
 
-        if (currentUsage < getHeapBytesThreshold() || currentUsage < allowedUsage) {
+        if (currentUsage < threshold || currentUsage < allowedUsage || HEAP_SIZE_BYTES == 0) {
```

Comment:
I feel that we should not enable this tracker (by not adding it to the list of trackers) if we cannot identify the memory limit. The current approach will just add a dummy heap usage tracker that is executed every time but never marks any task for cancellation.

Reply:
Good point. Initially I thought of doing the same but adding dummy heap usage tracker helps us get the heap usage stats. This serves our purpose of avoiding any cancellations which are not breaching the threshold and we can also track the heap usage.

Reply:
@nssuresh2007 can you please share your thoughts on this?

Reply:
As @ketanv3 mentioned above, I think we need not include the Heap Usage tracker when the heap max limit cannot be calculated and it can be removed from the stats API response as well.

Reply:
Thanks for your input @nssuresh2007. I have added commit to address this.